include/quicr/subscribe_track_handler.h

```diff
@@ -63,17 +63,20 @@ namespace quicr {
          *
          * @param full_track_name           Full track name struct
          * @param joining_fetch             If set, subscribe with a joining fetch using these attributes.
+         * @param publisher_initiated       True if publisher initiated the subscribe, otherwise False
          */
         SubscribeTrackHandler(const FullTrackName& full_track_name,
                               messages::SubscriberPriority priority,
                               messages::GroupOrder group_order,
                               messages::FilterType filter_type,
-                              const std::optional<JoiningFetch>& joining_fetch = std::nullopt)
+                              const std::optional<JoiningFetch>& joining_fetch = std::nullopt,
+                              bool publisher_initiated = false)
           : BaseTrackHandler(full_track_name)
           , priority_(priority)
           , group_order_(group_order)
           , filter_type_(filter_type)
           , joining_fetch_(joining_fetch)
+          , publisher_initiated_(publisher_initiated)
```

Comment:
Just an API design thing, but should we enforce the fact that a publisher initiated subscribe handler can't use joining fetch? Or maybe two constructors?

Reply:
Changed it to null opt joining fetch if provided when publisher initiated is true.